### PR TITLE
[feat]: [CDS-69600]: Addition of yaml field to environment group response

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
Addition of yaml field to environment group response

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50171)

### CODE REVIEW

**Comment by lovish1999** [comment]
trigger uts

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverrideRequestDTOV2.java [hunk 1]
```diff
@@ -46,5 +46,5 @@ public class ServiceOverrideRequestDTOV2 {
 
   @Schema(description = "Spec for overrides, containing overriding fields like manifests, variables, config files")
   ServiceOverridesSpec spec;
-  @Schema(description = "YAML for the Service Overrides Spec") String yamlInternal;
```
Comment (lovish1999) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverrideRequestDTOV2.java:49]: This change is done as part of another issue. We do not want to show yamlInternal field to user.

File: 120-ng-manager/src/test/resources/oas/exclusion-file [hunk 2]
```diff
@@ -1 +1,2 @@
-dto-exclusion: io.harness.ng.core.serviceoverridev2.beans.ServiceOverridesSpec
\ No newline at end of file
+dto-exclusion: io.harness.ng.core.serviceoverridev2.beans.ServiceOverridesSpec
+dto-exclusion: io.harness.ng.core.serviceoverridev2.beans.ServiceOverrideRequestDTOV2
```
Comment (lovish1999) [inline 120-ng-manager/src/test/resources/oas/exclusion-file:2]: This change is done as part of another issue. We do not want to show yamlInternal field to user.

**Comment by lovish1999** [comment]
trigger messagecheck

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50171)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Comment by yogesh-chauhan** [comment]
are we doing this for Terraform ?

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50171)* status: 4 file(s) to review. 1 of 2 approvals obtained. 

*[a discussion](https://reviewable.io/reviews/harness/harness-core/50171#-NZbWN6tCf1HQEVe1YMu:-NZdNC8m7XOpYrcrJ7nL:bfz0xm1) (no related file):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

are we doing this for Terraform ?
</blockquote></details>

The yaml Internal field was added for overrides v2 terraform in an another PR. This yaml field for environment group is also added for terraform as there was a bug reported that yaml is null when trying to use the terraform data source.

___

**Review by tchaurasiya — APPROVED** [review]

**Comment by lovish1999** [comment]
trigger review

**Pull request merged**